inference-engine/samples/speech_sample/main.cpp

```diff
@@ -1024,24 +1024,39 @@ int main(int argc, char *argv[]) {
                             continue;
                         }
 
+                        ptrInputBlobs.clear();
                         if (FLAGS_iname.empty()) {
-                            size_t num_files = FLAGS_iname.empty() ? numInputArkFiles : ptrInputBlobs.size();
-                            for (size_t i = 0; i < num_files; ++i) {
-                                MemoryBlob::Ptr minput = as<MemoryBlob>(ptrInputBlobs[i]);
-                                if (!minput) {
-                                    slog::err << "We expect ptrInputBlobs[" << i
-                                              << "] to be inherited from MemoryBlob, " <<
-                                              "but in fact we were not able to cast input blob to MemoryBlob"
-                                              << slog::endl;
-                                    return 1;
+                            for (auto &input : cInputInfo) {
+                                ptrInputBlobs.push_back(inferRequest.inferRequest.GetBlob(input.first));
+                            }
+                        } else {
+                            std::vector<std::string> inputNameBlobs = ParseBlobName(FLAGS_iname);
+                            for (const auto& input : inputNameBlobs) {
+                                Blob::Ptr blob = inferRequests.begin()->inferRequest.GetBlob(input);
+                                if (!blob) {
+                                    std::string errMessage("No blob with name : " + input);
+                                    throw std::logic_error(errMessage);
                                 }
-                                // locked memory holder should be alive all time while access to its buffer happens
-                                auto minputHolder = minput->wmap();
+                                ptrInputBlobs.push_back(blob);
+                            }
+                        }
 
-                                std::memcpy(minputHolder.as<void *>(),
-                                            inputFrame[i],
-                                            minput->byteSize());
+                        size_t num_files = FLAGS_iname.empty() ? numInputArkFiles : ptrInputBlobs.size();
```

Comment:
Is there a chance that numInputArkFiles != ptrInputBlobs.size() ?
IMO there is not.